src/sdk/ca-base/ca.ts

```diff
@@ -102,15 +105,17 @@ export class CA {
     onIntent: OnIntentHook;
     onSwapIntent: OnSwapIntentHook;
   } = {
-    onAllowance: (data) => data.allow(data.sources.map(() => 'min')),
-    onIntent: (data) => data.allow(),
-    onSwapIntent: (data) => data.allow(),
-  };
+      onAllowance: (data) => data.allow(data.sources.map(() => 'min')),
+      onIntent: (data) => data.allow(),
+      onSwapIntent: (data) => data.allow(),
+    };
   protected _initStatus = INIT_STATUS.CREATED;
   protected _networkConfig: NetworkConfig;
   protected _refundInterval: number | undefined;
   protected _initPromise: Promise<void> | null = null;
-  private readonly simulationClient: BackendSimulationClient;
+  private simulationClient: BackendSimulationClient;
+  protected _analytics?: AnalyticsManager; // Analytics manager set by subclass
```

Comment:
I am not sure why we use `AnalyticsManager | undefined`. here. Why not make this as a param of the constructor because this will be always set by subclass `NexusSDK` and having it as potentially undefined makes our lives just more miserable.